yb-voyager/src/queryparser/unsupportedConstructs.go

```diff
@@ -0,0 +1,162 @@
+from_clause:{range_subselect:{subquery:{select_stmt:{target_list:{res_target:{val:{func_call:{funcname:{string:{sval:"pg_advisory_lock"}}
+args:{param_ref:{number:1  location:39}}  funcformat:COERCE_EXPLICIT_CALL  location:22}}  location:22}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}
+alias:{aliasname:"lock_acquired"}}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsAdvisoryLocksInFromClause(fromClause []*pg_query.Node) bool {
+	for _, fromItem := range fromClause {
+		if subselectNode, isSubselect := fromItem.Node.(*pg_query.Node_RangeSubselect); isSubselect {
+			subSelectStmt := subselectNode.RangeSubselect.Subquery.GetSelectStmt()
+			if subSelectStmt != nil {
+				// Recursively check for advisory locks in the subquery's target list and FROM clause
+				if containsAdvisoryLocksInTargetList(subSelectStmt.TargetList) {
+					return true
+				}
+				if containsAdvisoryLocksInFromClause(subSelectStmt.FromClause) {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+/*
+Recursively checks the WHERE clause for advisory lock functions.
+
+Example: SELECT id, first_name FROM employees WHERE pg_try_advisory_lock($1) IS TRUE;
+stmts:{stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"id"}}  location:7}}  location:7}}
+target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"first_name"}}  location:11}}  location:11}}
+from_clause:{range_var:{relname:"employees"  inh:true  relpersistence:"p"  location:27}}
+where_clause:{boolean_test:{arg:{func_call:{funcname:{string:{sval:"pg_try_advisory_lock"}}  args:{param_ref:{number:1  location:64}}
+funcformat:COERCE_EXPLICIT_CALL  location:43}}  booltesttype:IS_TRUE  location:68}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsAdvisoryLocksInWhereClause(whereClause *pg_query.Node) bool {
+	if whereClause == nil {
+		return false
+	}
+
+	if funcCallNode := whereClause.GetFuncCall(); funcCallNode != nil {
+		funcList := funcCallNode.Funcname
+		functionName := funcList[len(funcList)-1].GetString_().Sval
+		if slices.Contains(advisoryLockFunctions, functionName) {
+			return true
+		}
+	}
+
+	// Recursively check for advisory locks in nested expressions
+	switch n := whereClause.Node.(type) {
+	case *pg_query.Node_SubLink:
+		/*
+			SELECT id, first_name FROM employees WHERE salary > $1 AND EXISTS (SELECT $2 FROM pg_advisory_lock($3))
+			subSelectStmt: target_list:{res_target:{val:{param_ref:{number:2 location:77}} location:77}}
+			from_clause:{range_function:{functions:{list:{items:{func_call:{funcname:{string:{sval:"pg_advisory_lock"}}
+			args:{param_ref:{number:3 location:102}} funcformat:COERCE_EXPLICIT_CAL
+		*/
+		if subSelectStmt := n.SubLink.Subselect.GetSelectStmt(); subSelectStmt != nil {
+			return containsAdvisoryLocksInTargetList(subSelectStmt.TargetList) ||
+				containsAdvisoryLocksInFromClause(subSelectStmt.FromClause)
+		}
```

Comment:
There can be possiblility of having the function call again in sub query's WHERE clause as well right?

Reply:
But I think it will be handled as part of complex queries task, right @sanyamsinghal ?

Reply:
Yes right, i tried to cover all complex cases, but it is taking time, so i will cover rest of them later.